pkg/didcomm/protocol/didexchange/service.go

```diff
@@ -78,11 +82,42 @@ type Service struct {
 	connectionStore *ConnectionRecorder
 }
 
+// IntroduceNotifier notifies the introduce protocol after receiving an invitation
+type IntroduceNotifier interface {
+	InvitationReceived(pid string) error
+}
+
 type context struct {
+	IntroduceNotifier
 	outboundDispatcher dispatcher.Outbound
 	signer             kms.Signer
 	connectionStore    *ConnectionRecorder
 	vdriRegistry       vdriapi.Registry
+	service            func(id string) (interface{}, error)
+}
+
+func (ctx *context) loadIntroduceNotifier() error {
```

Comment:
This seems backwards.

Why wouldn't Introduce register itself with DID Exchange?

Reply:
Do you mean just create a callback function?

Reply:
Yeah, it makes sense.

Reply:
Done!